src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/reports/EnvelopeCountSummaryRepository.java

```diff
@@ -15,18 +15,26 @@
     @Query(
         nativeQuery = true,
         value = "SELECT\n"
-            + "  container,\n"
-            + "  date,\n"
+            + "  first_events.container AS container,\n"
+            + "  first_events.date AS date,\n"
             + "  count(*) AS received,\n"
-            + "  SUM(CASE WHEN event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE') THEN 1 ELSE 0 END) AS rejected\n"
+            + "  SUM(CASE WHEN rejection_events.id IS NOT NULL THEN 1 ELSE 0 END) AS rejected\n"
             + "FROM (\n"
             + "  SELECT DISTINCT on (container, zipfilename)\n"
-            + "  container, zipfilename, date(createdat), event\n"
+            + "    container, zipfilename, date(createdat), event\n"
             + "  FROM process_events\n"
             + "  ORDER BY container, zipfilename, date(createdat) ASC\n"
             + ") AS first_events\n"
-            + "GROUP BY container, date\n"
-            + "HAVING date = :date"
+            + "LEFT JOIN (\n"
+            + "  SELECT DISTINCT on (container, zipfilename)\n"
+            + "    id, container, zipfilename\n"
+            + "  FROM process_events\n"
+            + "  WHERE event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE')\n"
+            + ") AS rejection_events\n"
+            + "ON rejection_events.container = first_events.container\n"
+            + "AND rejection_events.zipfilename = first_events.zipfilename\n"
+            + "GROUP BY first_events.container, first_events.date\n"
+            + "HAVING date = :date\n"
```

Comment:
function `date()` can simply go lastly here and all other dates will be dealt in full - keeping the correct order of events